src/Orleans/Runtime/GrainReference.cs

```diff
@@ -118,27 +128,37 @@ private GrainReference(GrainId grainId, string genericArgument, SiloAddress syst
         /// </summary>
         /// <param name="other">The reference to copy.</param>
         protected GrainReference(GrainReference other)
-            : this(other.GrainId, other.genericArguments, other.SystemTargetSilo, other.ObserverId) { }
+            : this(other.GrainId, other.genericArguments, other.SystemTargetSilo, other.ObserverId, other.RuntimeClient) { }
```

Comment:
since the `RuntimeClient` property throws if unbound, is this by design? or did you intend to copy the field instead without throwing?